p2p/raintree/network.go

```diff
@@ -24,43 +27,51 @@ var _ typesP2P.Network = &rainTreeNetwork{}
 type rainTreeNetwork struct {
 	base_modules.IntegratableModule
 
-	selfAddr       cryptoPocket.Address
-	pstoreProvider peerstore_provider.PeerstoreProvider
-
-	peersManager *rainTreePeersManager
-	nonceDeduper *mempool.GenericFIFOSet[uint64, uint64]
-
-	currentHeightProvider providers.CurrentHeightProvider
-
 	logger *modules.Logger
+	// host represents a libp2p network node, it encapsulates a libp2p peerstore
+	// & connection manager. `libp2p.New` configures and starts listening
+	// according to options.
+	// (see: https://pkg.go.dev/github.com/libp2p/go-libp2p#section-readme)
+	host libp2pHost.Host
+	// selfAddr is the pocket address representing this host.
+	selfAddr              cryptoPocket.Address
+	peersManager          *rainTreePeersManager
+	pstoreProvider        peerstore_provider.PeerstoreProvider
+	currentHeightProvider providers.CurrentHeightProvider
+	nonceDeduper          *mempool.GenericFIFOSet[uint64, uint64]
 }
 
-func NewRainTreeNetwork(addr cryptoPocket.Address, bus modules.Bus, pstoreProvider providers.PeerstoreProvider, currentHeightProvider providers.CurrentHeightProvider) typesP2P.Network {
+// TECHDEBT: refactor signature to receive a config struct
+// (i.e. `NewRainTreeNetwork(bus modules.Bus, cfg NetworkConfig).
+func NewRainTreeNetwork(host libp2pHost.Host, addr cryptoPocket.Address, bus modules.Bus, pstoreProvider providers.PeerstoreProvider, currentHeightProvider providers.CurrentHeightProvider) (typesP2P.Network, error) {
 	networkLogger := logger.Global.CreateLoggerForModule("network")
 	networkLogger.Info().Msg("Initializing rainTreeNetwork")
 
-	pstore, err := pstoreProvider.GetStakedPeerstoreAtHeight(currentHeightProvider.CurrentHeight())
-	if err != nil {
-		networkLogger.Fatal().Err(err).Msg("Error getting pstore")
+	if pstoreProvider == nil {
+		return nil, fmt.Errorf("peerstore provider required, got nil")
 	}
 
-	pm, err := newPeersManager(addr, pstore, true)
-	if err != nil {
-		networkLogger.Fatal().Err(err).Msg("Error initializing rainTreeNetwork rainTreePeersManager")
+	if currentHeightProvider == nil {
+		return nil, fmt.Errorf("current height provider required, got nil")
 	}
 
 	p2pCfg := bus.GetRuntimeMgr().GetConfig().P2P
 
 	n := &rainTreeNetwork{
+		host:                  host,
 		selfAddr:              addr,
-		peersManager:          pm,
 		nonceDeduper:          mempool.NewGenericFIFOSet[uint64, uint64](int(p2pCfg.MaxMempoolCount)),
 		pstoreProvider:        pstoreProvider,
 		currentHeightProvider: currentHeightProvider,
 		logger:                networkLogger,
 	}
 	n.SetBus(bus)
-	return typesP2P.Network(n)
+
+	if err := n.setupDependencies(); err != nil {
```

Comment:
I think we should have a consistent style on whether `setupDependencies` fatals or returns an error. No preference on which one as long as its consistent.

![Screenshot 2023-03-27 at 1 01 37 PM](https://user-images.githubusercontent.com/1892194/228053360-d6bec6a8-e806-497f-ad93-146f15938d57.png)

Reply:
:100: :+1: